tests/services/kernels/test_events.py

```diff
@@ -0,0 +1,78 @@
+import pytest
+from jupyter_client.manager import AsyncKernelManager
+from tornado import web
+
+from jupyter_server.services.kernels.kernelmanager import ServerKernelManager
+
+pytest_plugins = ["jupyter_events.pytest_plugin"]
+
+
+@pytest.mark.parametrize("action", ["start", "restart", "interrupt", "shutdown"])
+async def test_kernel_action_success_event(
+    monkeypatch, action, jp_read_emitted_events, jp_event_handler
+):
+    manager = ServerKernelManager()
+    manager.event_logger.register_handler(jp_event_handler)
```

Comment:
@Zsailer. I'm looking into adding event handling to the gateway tests and find this block of lines a little confusing and I'm not sure if this is a practical test.

If the _handler_ for an event is meant for an event's _consumer_ and the `ServerKernelManager` instance does not get instantiated until a kernel is started, how would a consumer be able to set their handler on _that_ instance?  It seems like handler registration can only take place on "singleton instances" - like `ContentManager` and `MappingKernelManager`, etc. where the instance is created via `init_configurables()`.

Hmm - I think I need to go RTD to understand the distinction between handlers and listeners.